src/plugins/interactive_setup/server/kibana_config/kibana_config_service.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { constants } from 'fs';
+import fs from 'fs/promises';
+import yaml from 'js-yaml';
+import path from 'path';
+
+import type { Logger } from 'src/core/server';
+
+import { getDetailedErrorMessage } from '../errors';
+
+export interface WriteConfigParameters {
+  host: string;
+  ca: string;
+  username: string;
+  password: string;
+}
+
+export class KibanaConfigService {
```

Comment:
Trying to work out what this service does. It's kind of just a wrapper to a config file isn't it? Maybe `ConfigWriter` or `ConfigFile` would be clearer?

Reply:
I renamed it to just `KibanaConfig` - the class that works with Kibana global configuration. Wanted to logically separate it from `Config` that is responsible for the plugin specific config. What do you think?

Reply:
Yeh, I get what you're trying to do but I wouldn't know what the difference is between Config and KibanaConfig if you wouldn't tell me. Config and ConfigFile / ConfigWriter or something else along those lines is clearer to me.

Reply:
Hmm, I see, from all the options `KibanaConfigWriter` would be the most technically correct, does it sound good? I want to keep `Kibana` or `Global` or something similar to clearly indicate that's not plugin config.

Reply:
Ah got you! Yeh let's use `KibanaConfigWriter` then.